docs/lang/articles/advanced/differentiable_programming.md

```diff
@@ -63,40 +73,18 @@ def compute_y():
 with ti.Tape(y):
     compute_y()
 
-print('dy/dx =', x.grad[None])
-print('at x =', x[None])
+print('dy/dx =', x.grad[None], ' at x =', x[None])
 ```
+### Case study: gravity simulation
 
-It's equivalant to:
-
-```python
-x = ti.field(float, ())
-y = ti.field(float, ())
-dy_dx = ti.field(float, ())
-
-@ti.kernel
-def compute_dy_dx():
-    dy_dx[None] = ti.cos(x[None])
-
-compute_dy_dx()
-
-print('dy/dx =', dy_dx[None])
-print('at x =', x[None])
-```
-
-### Usage example
-
-For a physical simulation, sometimes it could be easy to compute the
-energy but hard to compute the force on each particles.
-
-But recall that we can differentiate (negative) potential energy to get
-forces. a.k.a.: $F_i = -dU / dx_i$. So once you've write a kernel that
-is able to compute the potential energy, you may use Taichi's autodiff
-system to obtain the derivative of it and then the force on each
-particles.
+A common problem in physical simulation is that it's usually easy to compute
+energy but hard to compute force on every particle. Recall that we can differentiate
```

Comment:
Some compelling examples on this point for reference:
- [FEM with Lagrangian forces](https://github.com/taichi-dev/taichi/blob/master/examples/simulation/fem128.py)
- [Bond bending (and torsion) in molecular dynamics](https://github.com/victoriacity/taichimd/blob/5a44841cc8dfe5eb97de51f1d46f1bede1cc9936/taichimd/interaction.py#L190-L220)